hack/verify-deps.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env sh
+
+set -o errexit
+set -o nounset
+
+print_failure() {
+    cat <<EOF >&2
+There are unexpected changes to the vendor tree following 'go mod
+tidy' and 'go mod vendor'. Please run these commands locally and check
+the Git repository for any changes that may need to be committed.
+EOF
+    exit 1
+}
+
+if [ "${GITHUB_ACTIONS:-false}" = "true" ]; then
+    go mod tidy
+    go mod vendor
```

Comment:
can u replace the above with `make vendors` ?

Reply:
Not trivially. When you invoke `make vendors` within the container, the following happens:

```console
basename: missing operand
Try 'basename --help' for more information.
basename: missing operand
Try 'basename --help' for more information.
OCI_BIN_PATH=
OCI_BIN=
```

Reply:
I think I will pull `./hack/verify-deps.sh` out of this PR. It's now orthogonal to speeding up the build. We can rework the vendor verification in a separate PR.